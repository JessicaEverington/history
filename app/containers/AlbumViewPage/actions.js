/*
 *
 * AlbumViewPage actions
 *
 */

import {
  LOAD_ALBUM,
  LOAD_ALBUM_SUCCESS,
  LOAD_ALBUM_ERROR,
  LOAD_THUMB_LINKS_SUCCESS,
  LOAD_THUMB_LINKS_ERROR,
} from './constants';

function parseQueryString(find, from) {
  return RegExp(`[?&]${find}(=([^&#]*)|&|#|$)`).exec(from)[2];
}

export function loadAlbum(querystring, album) {
  return {
    type: LOAD_ALBUM,
    gallery: parseQueryString('gallery', querystring),
    album,
  };
}

export function albumLoaded(gallery, thumbs) {
  return {
    type: LOAD_ALBUM_SUCCESS,
    thumbs,
    gallery,
  };
}

export function albumLoadingError(error) {
  return {
    type: LOAD_ALBUM_ERROR,
    error,
  };
}

export function thumbLinksLoaded(thumbs) {
  return {
    type: LOAD_THUMB_LINKS_SUCCESS,
    thumbs,
  };
}

export function thumbLinksLoadingError(error) {
  return {
    type: LOAD_THUMB_LINKS_ERROR,
    error,
  };
}
