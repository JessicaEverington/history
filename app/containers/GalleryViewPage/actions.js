/*
 *
 * GalleryViewPage actions
 *
 */

import {
  LOAD_GALLERY,
  LOAD_GALLERY_SUCCESS,
  LOAD_GALLERY_ERROR,
} from './constants';

export function loadGallery(galleryName) {
  return {
    type: LOAD_GALLERY,
    galleryName,
  };
}

export function galleryLoaded(galleryXml) {
  return {
    type: LOAD_GALLERY_SUCCESS,
    galleryXml,
  };
}

export function galleryLoadingError(error) {
  return {
    type: LOAD_GALLERY_ERROR,
    error,
  };
}

