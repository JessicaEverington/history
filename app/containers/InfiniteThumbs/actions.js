import {
  CHOOSE_MEMORY,
} from './constants';

export function chooseMemory(id) {
  return {
    type: CHOOSE_MEMORY,
    id,
  };
}
