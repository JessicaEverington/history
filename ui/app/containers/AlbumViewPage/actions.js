import {
  LOAD_ALBUM,
  LOAD_ALBUM_SUCCESS,
  LOAD_ALBUM_ERROR,
  LOAD_NEXT_THUMB_PAGE,
  LOAD_NEXT_THUMB_PAGE_SUCCESS,
  LOAD_NEXT_THUMB_PAGE_ERROR,
  LOAD_THUMBS_SUCCESS,
  NEXT_MEMORY,
  PREV_MEMORY,
} from './constants';

function parseQueryString(find, from) {
  return RegExp(`[?&]${find}(=([^&#]*)|&|#|$)`).exec(from)[2];
}

export function loadAlbum(querystring, album) {
  return {
    type: LOAD_ALBUM,
    gallery: parseQueryString('gallery', querystring),
    album,
  };
}

export function albumLoadSuccess({ gallery, album, memories }) {
  return {
    type: LOAD_ALBUM_SUCCESS,
    gallery,
    album,
    memories,
  };
}

export function albumLoadError(error) {
  return {
    type: LOAD_ALBUM_ERROR,
    error,
  };
}

export function loadNextPage() {
  return {
    type: LOAD_NEXT_THUMB_PAGE,
  };
}

export function nextPageSuccess({ gallery, album, newMemories, page }) {
  return {
    type: LOAD_NEXT_THUMB_PAGE_SUCCESS,
    gallery,
    album,
    newMemories,
    page,
    hasMore: true,
  };
}

export function nextPageError(error) {
  return {
    type: LOAD_NEXT_THUMB_PAGE_ERROR,
    error,
  };
}

export function thumbsLoaded({ gallery, album, newMemories, page }) {
  return {
    type: LOAD_THUMBS_SUCCESS,
    gallery,
    album,
    newMemories,
    page,
    hasMore: false,
  };
}

export function chooseAdjacentMemory(adjacentInt) {
  if (adjacentInt > 0) {
    return {
      type: NEXT_MEMORY,
      adjacentInt,
    };
  }

  return {
    type: PREV_MEMORY,
    adjacentInt,
  };
}
