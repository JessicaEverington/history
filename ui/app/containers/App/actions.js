import {
  CHOOSE_MEMORY,
  LOAD_PHOTO_ERROR,
  LOAD_PHOTO_SUCCESS,
} from './constants';

export function chooseMemory(id) {
  console.log('action chooseMemory; id ', id);
  return {
    type: CHOOSE_MEMORY,
    id,
  };
}

export function photoLoadSuccess({
  gallery, album, id, photoLink,
}) {
  console.log('action photoLoadSuccess');
  return {
    type: LOAD_PHOTO_SUCCESS,
    gallery,
    album,
    id,
    photoLink,
  };
}

export function photoLoadError(error) {
  return {
    type: LOAD_PHOTO_ERROR,
    error,
  };
}
