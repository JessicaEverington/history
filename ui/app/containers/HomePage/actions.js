import {
  LOAD_GALLERIES,
  LOAD_GALLERIES_SUCCESS,
  LOAD_GALLERIES_ERROR,
} from './constants';

export function loadGalleries() {
  return {
    type: LOAD_GALLERIES,
  };
}

export function galleriesLoaded(galleries) {
  return {
    type: LOAD_GALLERIES_SUCCESS,
    galleries,
  };
}

export function galleriesLoadingError(error) {
  return {
    type: LOAD_GALLERIES_ERROR,
    error,
  };
}
