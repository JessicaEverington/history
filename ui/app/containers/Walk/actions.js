import {
  LIST_DIRECTORY_REQUEST,
  LIST_DIRECTORY_SUCCESS,
  LIST_DIRECTORY_FAILURE,
} from './constants';

function getFilesByPath() {
  return {
    type: LIST_DIRECTORY_REQUEST,
  };
}

function listingSuccess(listing) {
  return {
    type: LIST_DIRECTORY_SUCCESS,
    listing,
  };
}

function listingFailure(error) {
  return {
    type: LIST_DIRECTORY_FAILURE,
    error,
  };
}

export default {
  getFilesByPath,
  listingSuccess,
  listingFailure,
};
