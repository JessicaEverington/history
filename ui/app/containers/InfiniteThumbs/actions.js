import { LOAD_THUMBS_SUCCESS } from '../AlbumViewPage/constants';

export function thumbsLoaded({
  gallery,
  album,
  newMemories,
  page,
}) {
  return {
    type: LOAD_THUMBS_SUCCESS,
    gallery,
    album,
    newMemories,
    page,
    hasMore: false,
  };
}
