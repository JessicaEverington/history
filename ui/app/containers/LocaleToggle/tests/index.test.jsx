/* global beforeAll, describe, jest, expect, test */
import React from 'react';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { render } from 'react-testing-library';

import LocaleToggleComponent, { mapDispatchToProps } from '../index';
import { changeLocale } from '../../LanguageProvider/actions';
import LanguageProviderComponent from '../../LanguageProvider';

import configureStore from '../../../configureStore';
import { translationMessages } from '../../../i18n';

describe('<LocaleToggle />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  test('should match the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <LanguageProviderComponent messages={translationMessages}>
          <LocaleToggleComponent />
        </LanguageProviderComponent>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should present the default `en` english language option', () => {
    const { container } = render(
      <Provider store={store}>
        <LanguageProviderComponent messages={translationMessages}>
          <LocaleToggleComponent />
        </LanguageProviderComponent>
      </Provider>,
    );
    expect(container.querySelector('option[value="en"]')).not.toBeNull();
  });

  describe('mapDispatchToProps', () => {
    describe('onLocaleToggle', () => {
      test('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onLocaleToggle).toBeDefined();
      });

      test.skip('should dispatch changeLocale when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const locale = 'de';
        const evt = { target: { value: locale } };
        result.onLocaleToggle(evt);
        expect(dispatch).toHaveBeenCalledWith(changeLocale(locale));
      });
    });
  });
});
