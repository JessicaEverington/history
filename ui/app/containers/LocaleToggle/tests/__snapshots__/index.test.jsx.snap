// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocaleToggle /> should match the snapshot 1`] = `
<div
  class="Wrapper-ubnoo8-0 iGcvVP"
>
  <select
    class="Select-sc-19ubqap-0 izskqM"
  >
    <option
      value="en"
    >
      en
    </option>
    <option
      value="de"
    >
      de
    </option>
  </select>
</div>
`;
